spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java

```diff
@@ -63,7 +63,9 @@ public static void startMetastoreAndSpark() {
 
   @AfterClass
   public static void stopMetastoreAndSpark() {
-    catalog.close();
+    if (catalog != null) {
+      catalog.close();
+    }
```

Comment:
Can we create a separate PR for this and merge it in sooner? I imagine this PR will take longer to merge because of its large scope, but this seems useful now.

Reply:
Sure, I just added this in because my tests were yelling at me all the time because I like to throw random kill codes around when debugging. I decided to just make sure it would stop bothering me.

Reply:
I can create a separate PR to handle this now if you don't mind.

Reply:
@RussellSpitzer I created a PR that cherry picks just this change as this PR's scope is pretty large and this small change can be merged in much more quickly: https://github.com/apache/iceberg/pull/1529

Reply:
#1529 was merged.